src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -434,17 +436,17 @@ impl<T: 'static> EventLoop<T> {
                 if let Ok(dispatched) = queue.dispatch_pending(state, |_, _, _| unimplemented!()) {
                     dispatched > 0
                 } else {
-                    break;
+                    break 1;
                 }
             };
 
             match control_flow {
-                ControlFlow::Exit => break,
+                ControlFlow::ExitWithCode(code) => break code,
                 ControlFlow::Poll => {
                     // Non-blocking dispatch.
                     let timeout = Duration::from_millis(0);
                     if self.loop_dispatch(Some(timeout)).is_err() {
-                        break;
+                        break 1;
```

Comment:
Please, return last os error code since the only way we could fail due dispatch if something is wrong with polling it.  And I think there's a way to get code from error enum itself, but I'm not sure if it's still there, but it should.

Reply:
I'll take a look at it, thanks!

Reply:
Yeah, it should. Since it's io::Error under the hood, so just getting code from it should work.

Reply:
From what I see in https://doc.rust-lang.org/stable/std/io/struct.Error.html#method.raw_os_error, the underlying OS error code can only be fetched by using `std::io::Error::last_os_error`, whose usages I can neither find in winit (well, on windows, but that's not relevant here) nor in `calloop`s code. Was calling `std::io::Error::last_os_error` to get an error to examine for the exit code in your mind or did you mean something else?

Reply:
No, I mean that `self.loop_dispatch` is returning `std::io::Result` from which you can get `std::io::Error` and on which you can call `https://doc.rust-lang.org/std/io/struct.Error.html#method.raw_os_error`, since the `Error` it's using under the hood is `nix` Error, but it's being built from `errno` value and its from impl is using `from_raw_os_error` when converting.

Reply:
Fair enough. Thanks for the explanation!

Reply:
>nor in calloops code, so the error from self.loop_dispatch likely does not contain the error code. 

It does contain it, since that's what `epoll_wait` returns in calloop.

Reply:
There we go, are the changes fine for you?